Use a YAML file to store product information

mcdonnnj opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

This pull request is focused on switching to using a YAML file to store information about affected products. This is achieved through the following:

- Use the existing `SOFTWARE-LIST.md` contents to populate an appropriately formatted YAML file.
- Add a GitHub Actions workflow that will update the contents of `SOFTWARE-LIST.md` when a `push` event occurs on the `develop` branch. This will use the current contents of `data/cisagov.yml` (the YAML file) to populate `SOFTWARE-LIST.md`.
<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

## 💭 Motivation and context ##

Using a machine readable file with stricter formatting requirements to store this information should reduce human error and also make it easier for third parties to ingest our data.
<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such -->
<!-- as "closes" or "resolves" to auto-close them on merge. -->

## 🧪 Testing ##

Automated tests pass. I tested this workflow configuration in [this branch of a testing repository](https://github.com/cisagov/github-actions-playground/tree/testing/log4j_list_update) and it worked as expected.
<!-- How did you test your changes? How could someone else test this PR? -->
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

<!--
## 📷 Screenshots (if appropriate) ##

Uncomment this section if a screenshot is needed.

-->

## ✅ Pre-approval checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] All new and existing tests pass.

dav3r committed:
Add template for README

dav3r committed:
Merge branch 'develop' into improvement/add-readme-template

dav3r committed:
Merge branch 'develop' into improvement/add-readme-template

dav3r committed:
Rename README.md.tpl to SOFTWARE-LIST.md.tpl and update contents accordingly

mcdonnnj committed:
Make sure the template will get tested by linters

Change the template's extension to `.md` to ensure that it is checked
by the markdownlint hook (and any others).

mcdonnnj committed:
Add GitHub Actions workflow to update software list

Add a GitHub Actions workflow and related files to automatically update
SOFTWARE-LIST.md when a push to the develop branch occurs. This
leverages the cisagov/md-table-to-yaml Python library to perform the
conversions.

mcdonnnj committed:
Add initial cisagov data source

This follows the expected YAML format, but there are no software
records.

mcdonnnj committed:
Merge branch 'develop' into improvement/add-readme-template

mcdonnnj committed:
Update software information

Read the current contents of SOFTWARE-LIST.md and use them to populate
cisagov.yml with useful data. This is necessary for the switch to using
a YAML file (cisagov.yml) as the source of information in this
repository.

mcdonnnj committed:
Merge branch 'develop' into improvement/add-readme-template

mcdonnnj committed:
Remove product status from issue forms

This value is derived from other information in each software product
entry in the YAML file. Since it is now a derived value we should not
prompt submitters to provide one.

mcdonnnj committed:
Update requirements for the update software workflow

Update the workflow's requirements to use a specific version at the updated
location of the Python project doing the heavy lifting. Additionally the
requirements file is added to the actions/cache key used.

mcdonnnj committed:
Merge branch 'develop' into improvement/add-readme-template

mcdonnnj committed:
Update the cisagov YAML file

Update to reflect the current state of SOFTWARE-LIST.md.

mcdonnnj committed:
Change the list update workflow testing configuration

Update the testing branch for the software list update workflow to
include the SHA of the commit that triggers the workflow. This should
help track down problems if there is a failure in testing/rendering.

mcdonnnj committed:
Merge branch 'develop' into improvement/add-readme-template

mcdonnnj committed:
Update cisagov YAML with the latest software list contents

justmurphy reviewed (approved):
Looks great!

dav3r commented on .github/workflows/update_software_list.yml:
```diff
@@ -0,0 +1,111 @@
+          git merge origin/${{ needs.setup.outputs.testing_branch }}
+          git push
+      - name: Cleanup testing branch
+        run: git push -d origin ${{ needs.setup.outputs.testing_branch }}
```
Prefer full name of switch:
```suggestion
        run: git push --delete origin ${{ needs.setup.outputs.testing_branch }}
```

mcdonnnj replied:
Good catch!

dav3r reviewed (approved):
This looks fantastic to me - excellent work as always! 💪 

I mentioned one tiny thing to change, but otherwise it's great.

mcdonnnj committed:
Use the long form of a switch for git

We prefer using the long form of switches for command line programs to
improve maintainability and to better convey what is happening even if
someone is unfamiliar with the switches for a given program.

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>
